botfront/imports/ui/components/stories/StoryEditorContainer.jsx

```diff
@@ -84,9 +84,20 @@ const StoryEditorContainer = ({
             () => {},
             content => saveStory(story._id, { story: content }),
             templates,
+            story.checkpoints && story.checkpoints.length > 0,
         ),
     });
 
+    useEffect(() => {
+        if (storyControllers[story._id]) {
+            const change = storyControllers[story._id].isABranch !== (story.checkpoints && story.checkpoints.length > 0);
+            storyControllers[story._id].isABranch = story.checkpoints && story.checkpoints.length > 0;
+            if (change) {
+                storyControllers[story._id].validateStory();
+            }
+        }
+    }, [story.checkpoints && story.checkpoints.length]);
```

Comment:
If you have `[story.checkpoints && story.checkpoints.length]` as a second argument of useEffect, you already know `change` is true.